### DISCUSSION TITLE
Trouble using dp converted model

### DISCUSSION THREAD

**Discussion by rajnichahal** [deepmodeling/deepmd-kit#1981] [Q&A]
Hi, I trained a deepmd model using deepmd version 1.2.1. In order to use the python API for deepmd version 2.1.1, I converted the model using following:
dp convert-from 1.2 -i old_frozen_model.pb -o new_frozen_model.pb

The model conversion was successful, but it is giving me the following error message when using Python API as below:
dp = DeepPot('new_frozen_model.pb')

**Error message**
/usr/local/envs/deepmd/lib/python3.10/site-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.16.5 and <1.23.0 is required for this version of SciPy (detected version 1.23.1
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
WARNING:tensorflow:From /usr/local/envs/deepmd/lib/python3.10/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.
Instructions for updating:
non-resource variables are not supported in the long term
WARNING:root:To get the best performance, it is recommended to adjust the number of threads by setting the environment variables OMP_NUM_THREADS, TF_INTRA_OP_PARALLELISM_THREADS, and TF_INTER_OP_PARALLELISM_THREADS.
WARNING:root:Environment variable KMP_BLOCKTIME is empty. Use the default value 0
WARNING:root:Environment variable KMP_AFFINITY is empty. Use the default value granularity=fine,verbose,compact,1,0
/usr/local/envs/deepmd/lib/python3.10/importlib/__init__.py:169: UserWarning: The NumPy module was reloaded (imported a second time). This can in some cases result in small but subtle issues and is discouraged.
  _bootstrap._exec(spec, module)
Traceback (most recent call last):
  File "/project/uml_stephen_lam/Rajni/pca_tsne/deepmd_data/get_data3_11_atomic.py", line 9, in <module>
    dp = DeepPot('new_frozen_model.pb')
  File "/usr/local/envs/deepmd/lib/python3.10/site-packages/deepmd/infer/deep_pot.py", line 88, in __init__
    DeepEval.__init__(
  File "/usr/local/envs/deepmd/lib/python3.10/site-packages/deepmd/infer/deep_eval.py", line 41, in __init__
    self.graph = self._load_graph(
  File "/usr/local/envs/deepmd/lib/python3.10/site-packages/deepmd/infer/deep_eval.py", line 154, in _load_graph
    graph_def.ParseFromString(f.read())
google.protobuf.message.DecodeError: Error parsing message with type 'tensorflow.GraphDef'

Please let me know how can I resolve this error. Thanks!

**Comment by njzjz** [comment]
Could you provide the model?

**Reply by rajnichahal** [reply]
Here is the frozen model trained using version 1.2:
[https://studentuml-my.sharepoint.com/:u:/r/personal/rajni_chahal_uml_edu/Documents/DeePMD/[Flinazr_train7_2.pb](https://studentuml-my.sharepoint.com/:u:/r/personal/rajni_chahal_uml_edu/Documents/DeePMD/Flinazr_train7_2.pb?csf=1&web=1&e=DrSRdn)?csf=1&web=1&e=DrSRdn](url)

Here is the converted model to be compatible with version 2.1:
[https://studentuml-my.sharepoint.com/:u:/r/personal/rajni_chahal_uml_edu/Documents/DeePMD/[Flinazr_train7_2_v2.pb](https://studentuml-my.sharepoint.com/:u:/r/personal/rajni_chahal_uml_edu/Documents/DeePMD/Flinazr_train7_2_v2.pb?csf=1&web=1&e=8T1LBH)?csf=1&web=1&e=8T1LBH](url)

**Reply by njzjz** [reply]
Sorry, I can't access your files. It requires login.

**Reply by rajnichahal** [reply]
My apologies.
Here is the frozen model trained using version 1.2:
[Flinazr_train7_2.zip](https://github.com/deepmodeling/deepmd-kit/files/9757317/Flinazr_train7_2.zip)

Here is the converted model to be compatible with version 2.1:
[Flinazr_train7_2_v2.zip](https://github.com/deepmodeling/deepmd-kit/files/9757318/Flinazr_train7_2_v2.zip)

**Reply by njzjz** [reply]
The graph file in [Flinazr_train7_2_v2.zip](https://github.com/deepmodeling/deepmd-kit/files/9757318/Flinazr_train7_2_v2.zip) is a pbtxt file (raw text format). How did you generate it?

[`deepmd.utils.convert.convert_pbtxt_to_pb`](https://docs.deepmodeling.com/projects/deepmd/en/v2.1.5/api_py/deepmd.utils.html#deepmd.utils.convert.convert_pbtxt_to_pb) can be used to convert the pbtxt to pb format.

**Reply by rajnichahal** [reply]
Thanks for letting me know. I will try converting the file.

Every time I am using 'dp convert' to convert from version 1.2 to 2.1, I am getting .pbtxt file written instead of .pb.

**Reply by njzjz** [reply]
Sorry but I can't reproduce this behavior. What version do you use?

**Reply by rajnichahal** [reply]
Oh ok. The model was trained using version 1.3.3. I am converting to be compatible with version 2.1.1

**Reply by njzjz** [reply]
I can't get what is wrong.  Does your `convert_pbtxt_to_pb` work fine?

**Reply by rajnichahal** [reply]
1. Oh ok. Even if I use the Python API to convert, it is still writing .pbtxt file upon conversion:
import os
import textwrap
from deepmd.env import tf
from deepmd.utils import convert

convert.convert_13_to_21('Flinazr_train7_2.pb', 'output_model.pb')

2. Also, trying to convert .pbtxt to .pb as following--it is not qriting the output .pb file:
import os
import textwrap
from deepmd.env import tf
from deepmd.utils import convert

convert.convert_pbtxt_to_pb('frozen_model.pbtxt','new model.pb')


Please let me know if there is anything I am missing with convert_pbtxt_to_pb. Thanks

**Reply by njzjz** [reply]
If you never modified the following code (as_text=False), there may be a bug in the TensorFlow...

https://github.com/deepmodeling/deepmd-kit/blob/6e3d4a626af965e951298f1bce9a9d0a2bbda317/deepmd/utils/convert.py#L156

**Reply by rajnichahal** [reply]
Hi, I did not modify the code.

Please let me know what should be my option here.

**Reply by njzjz** [reply]
If you get the same result by manually executing the following tensorflow codes, you can consider submitting an issue to [tensorflow](https://github.com/tensorflow/tensorflow/issues). There is nothing we can do here.

https://github.com/deepmodeling/deepmd-kit/blob/6e3d4a626af965e951298f1bce9a9d0a2bbda317/deepmd/utils/convert.py#L151-L156

**Reply by rajnichahal** [reply]
Hi, 
I figured that my script was being killed and nothing was written because of less memory available. Converting as following is working now:
import os
import textwrap
from deepmd.env import tf
from deepmd.utils import convert
convert.convert_13_to_21('Flinazr_train7_2.pb', 'output_model.pb')

thanks for your help!
